ports/nrf/modules/machine/temp.c

```diff
@@ -79,16 +79,23 @@ STATIC mp_obj_t machine_temp_make_new(const mp_obj_type_t *type, size_t n_args,
 /// \method read()
 /// Get temperature.
 STATIC mp_obj_t machine_temp_read(mp_uint_t n_args, const mp_obj_t *args) {
+    int32_t temp;
 
 #if BLUETOOTH_SD
     if (BLUETOOTH_STACK_ENABLED() == 1) {
-        int32_t temp;
         (void)sd_temp_get(&temp);
         return MP_OBJ_NEW_SMALL_INT(temp / 4); // resolution of 0.25 degree celsius
     }
 #endif // BLUETOOTH_SD
 
-    return MP_OBJ_NEW_SMALL_INT(nrf_temp_read());
+    nrf_temp_event_clear(NRF_TEMP, NRF_TEMP_EVENT_DATARDY);
+    nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_START);
+    while (!nrf_temp_event_check(NRF_TEMP, NRF_TEMP_EVENT_DATARDY)) {
+    }
+    temp = nrf_temp_read();
+    nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_STOP);
```

Comment:
@Xenoamor, Sorry for moving forth and back on this one. =)

You are absolutely right. This STOP is not needed. I checked it up a bit more closely, and the intention of STOP is to kill a START in case you want to shut down the peripheral without the sample. It's one-shot as you say. So i think we can revert this STOP part back to the original proposal as our intention in this code is to get the sample.